src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+
+          var delta = deltaWithSettings(siteSettings)
+          delta = deltaWithWatchlist(langToWatchlist, delta)
+          delta = deltaWithBlacklist(blacklist, delta)
+
+          // Create transformContext and broadcast
+          updateTransformContext(delta, sparkContext)
+
+          // Start async update listener
+          startQueueClient()
+        }
+      }
+      finally {
+        writeLock.unlock()
+      }
+    }
+  }
+
+  private def startQueueClient(): Unit = {
+    queueClient = new QueueClient(
+      new ConnectionStringBuilder(
+        Properties.envOrElse("FORTIS_SERVICEBUS_NAMESPACE", ""),
+        Properties.envOrElse("FORTIS_SERVICEBUS_CONFIG_QUEUE", ""),
+        Properties.envOrElse("FORTIS_SERVICEBUS_POLICY_NAME", ""),
+        Properties.envOrElse("FORTIS_SERVICEBUS_POLICY_KEY", "")
+      ), ReceiveMode.PeekLock)
+
+    queueClient.registerMessageHandler(
+      new MessageHandler(),
+      new MessageHandlerOptions(
+        1, // Max concurrent calls
+        true,
+        Duration.ofMinutes(5)))
+  }
+
+  /**
+    * Creates a new transform context from the current one, overwriting any corresponding fields that are
+    * non-empty in the provided delta. Broadcast fields are broadcasted here as well, as needed.
+    *
+    * Note: this is only called by Spark threads.
+    */
+  private def updateTransformContext(delta: Delta, sparkContext: SparkContext): Unit = {
+    if (transformContext == null) {
+      transformContext = TransformContext()
+    }
+
+    transformContext = transformContext.copy(
+      siteSettings = delta.siteSettings.getOrElse(transformContext.siteSettings),
+      langToWatchlist = delta.langToWatchlist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.langToWatchlist
+      },
+      blacklist = delta.blacklist match {
+        case Some(list) => sparkContext.broadcast(list)
+        case None => transformContext.blacklist
+      },
+      locationsExtractorFactory = delta.locationsExtractorFactory match {
+        case Some(factory) => sparkContext.broadcast(factory)
+        case None => transformContext.locationsExtractorFactory
+      },
+      imageAnalyzer = delta.imageAnalyzer.getOrElse(transformContext.imageAnalyzer),
+      languageDetector = delta.languageDetector.getOrElse(transformContext.languageDetector),
+      sentimentDetectorAuth = delta.sentimentDetectorAuth.getOrElse(transformContext.sentimentDetectorAuth)
+    )
+  }
+
+  /**
+    * Builds a delta against the current transform context using the provided site settings.
+    * @param delta An existing delta to be used as a base.
+    * @return
+    */
+  private def deltaWithSettings(siteSettings: SiteSettings, delta: Delta = Delta()): Delta = {
+    // Note that parameters are call-by-name
+    def updatedField[T](isDirty: => Boolean, newVal: => T): Option[T] = {
+      // Invoke 'isDirty' only if transform context is not null
```

Comment:
That's not semantically equivalent. The idea is that `isDirty` is not evaluated unless the transform context is not null. I'm taking advantage of the early out of ||. Also note that `isDirty` is a named expression, and not invokable (). If this is too cryptic, I can rework or improve the comment.